src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/DoNotPrefixEnumValuesWithTypeNameTests.cs

```diff
@@ -139,6 +140,144 @@ enum State
+        public async Task AllValuesPrefixed_Diagnostic(string editorConfigText)
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+                class A
+                {
+                    enum State
+                    {
+                        StateOk = 0,
+                        StateError = 1,
+                        StateUnknown = 2,
+                        StateInvalid = 3
+                    };
+                }"
+                    },
+                    AdditionalFiles = { (".editorconfig", editorConfigText) },
+                    ExpectedDiagnostics =
+                    {
+                        GetCSharpResultAt(6, 25, "State"),
+                        GetCSharpResultAt(7, 25, "State"),
+                        GetCSharpResultAt(8, 25, "State"),
+                        GetCSharpResultAt(9, 25, "State"),
+                    }
+                }
+            }.RunAsync();
+
+            await new VerifyVB.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+                Class A
+                    Enum State
+                        StateOk = 0
+                        StateError = 1
+                        StateUnknown = 2
+                        StateInvalid = 3
+                    End Enum
+                End Class
+"
+                    },
+                    AdditionalFiles = { (".editorconfig", editorConfigText) },
+                    ExpectedDiagnostics =
+                    {
+                        GetBasicResultAt(4, 25, "State"),
+                        GetBasicResultAt(5, 25, "State"),
+                        GetBasicResultAt(6, 25, "State"),
+                        GetBasicResultAt(7, 25, "State"),
+                    }
+                }
+            }.RunAsync();
+        }
+
+        [Theory]
+        // No data
+        [InlineData("")]
+        // Invalid option
+        [InlineData("dotnet_code_quality.CA1712.enum_values_prefix_trigger = FOO")]
+        // Valid options
+        [InlineData("dotnet_code_quality.CA1712.enum_values_prefix_trigger = AnyEnumValue")]
+        [InlineData("dotnet_code_quality.CA1712.enum_values_prefix_trigger = AllEnumValues")]
+        [InlineData("dotnet_code_quality.CA1712.enum_values_prefix_trigger = Heuristic")]
+        public async Task OneOfFourValuesPrefixed_Diagnostic(string editorConfigText)
+        {
+            var csharpTest = new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+                class A
+                {
+                    enum State
+                    {
+                        StateOk = 0,
```

Comment:
You can probably add one more test where 3 out of the 4 fields have the prefix and validate the diagnostic fires for heuristic and any enum cases, but not the all enum values case.